pkg/core/capability/manual/manual.go

```diff
@@ -0,0 +1,85 @@
+package manual
+
+import (
+	"errors"
+	"reflect"
+	"soarca/internal/logger"
+	"soarca/pkg/core/capability"
+	"soarca/pkg/interaction"
+	"soarca/pkg/models/cacao"
+	"soarca/pkg/models/execution"
+	"time"
+)
+
+var (
+	component = reflect.TypeOf(ManualCapability{}).PkgPath()
+	log       *logger.Log
+)
+
+const (
+	manualResultVariableName = "__soarca_manual_result__"
+	manualCapabilityName     = "soarca-manual-http"
+	fallbackTimeout          = time.Minute * 1
+)
+
+func New(controller interaction.ICapabilityInteraction,
+	channel chan interaction.InteractionResponse) ManualCapability {
+	// channel := make(chan interaction.InteractionResponse)
+	return ManualCapability{interaction: controller, channel: channel}
+}
+
+func init() {
+	log = logger.Logger(component, logger.Info, "", logger.Json)
+}
+
+type ManualCapability struct {
+	interaction interaction.ICapabilityInteraction
+	channel     chan interaction.InteractionResponse
+}
+
+func (manual *ManualCapability) GetType() string {
+	return manualCapabilityName
+}
+
+func (manual *ManualCapability) Execute(
+	metadata execution.Metadata,
+	context capability.Context) (cacao.Variables, error) {
+
+	command := interaction.InteractionCommand{Metadata: metadata, Context: context}
+
+	err := manual.interaction.Queue(command, manual.channel)
+	if err != nil {
+		return cacao.NewVariables(), err
+	}
+
+	result, err := manual.awaitUserInput(manual.getTimeoutValue(context.Step.Timeout))
+	if err != nil {
+		return cacao.NewVariables(), err
+	}
+	return result.Select(context.Step.OutArgs), nil
+
+}
+
+func (manual *ManualCapability) awaitUserInput(timeout time.Duration) (cacao.Variables, error) {
```

Comment:
@MaartendeKruijf by what I am implementing, I think using the context would be better also because we can also pass it to the Queue(function), capture the timeout inside the interaction object, and thus handle the idle-wait on the go routines Notify for integrations response more gracefully. Right?